src/Utility/Security.php

```diff
@@ -121,17 +121,35 @@ public static function randomBytes(int $length): string
      * Creates a secure random string.
      *
      * @param int $length String length. Default 64.
+     * @param bool $includeSpecialChars If to include special characters. Default false.
      * @return string
      */
-    public static function randomString(int $length = 64): string
+    public static function randomString(int $length = 64, bool $includeSpecialChars = false): string
     {
-        return substr(
-            bin2hex(Security::randomBytes((int)ceil($length / 2))),
+        $string = '';
+
+        if ($length === 0) {
+            return $string;
+        }
+
+        if ($includeSpecialChars) {
+            $special_charset = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~';
+
+            for ($i = 0; $i < rand(1, $length); $i++) {
+                $string .= $special_charset[random_int(0, strlen($special_charset))];
+            }
+        }
+
+        $string .= substr(
+            bin2hex(Security::randomBytes((int)ceil(($length - strlen($string)) / 2))),
             0,
-            $length
+            ($length - strlen($string))
         );
+
+        return str_shuffle($string);
```

Comment:
Shouldn't we be shuffling before  the substr if we're just prepending the special characters?